### ISSUE TITLE
`ostree config set` fails when setting negative values

### ISSUE THREAD

**Issue by r4f4** [ostreedev/ostree#1893]
If you try to set a negative value for a config option (e.g `lock-timeout-secs` with value `-1` means `LOCK_DISABLED`) you'll get instead:
```
./ostree config --repo=/tmp/test-repo-6Zwkyk/ set core.lock-timeout-secs -1
error: Unknown option -1
```
So `-1` is being considered a new command line option instead of a value for the just entered one.

**Comment by r4f4** [comment]
This issue comes from `g_option_context_parse` in `ot-main.c:348` not recognizing `-1` as a parameter for `config set` but as an option instead.

**Comment by rfairley** [comment]
Something else that's interesting is trying to turn off GLib parsing for the last argument with `ostree --repo=repo config set core.lock-timeout-secs -- -1` reports `error: KEY and VALUE must be specified`, suggesting the last argument is not being included in the `argc`/`argv` arguments (also does the same with a positive integer e.g. `2`). Would have expected this to be a workaround.

**Comment by r4f4** [comment]
The problem is the pre-parsing and re-arranging of the global options in `src/ostree/ot-main.c:142`. You can see here how it's removed from the command line:
```
# ./ostree --repo=/tmp/test-repo-qgO9cY/ config set core.payload-link-threshold -- -2
main argc=7
main argv[0] = /srv/ostree/.libs/lt-ostree
main argv[1] = --repo=/tmp/test-repo-qgO9cY/
main argv[2] = config
main argv[3] = set
main argv[4] = core.payload-link-threshold
main argv[5] = --
main argv[6] = -2
main argc=4
main argv[0] = /srv/ostree/.libs/lt-ostree
main argv[1] = --repo=/tmp/test-repo-qgO9cY/
main argv[2] = set
main argv[3] = core.payload-link-threshold
```

**Comment by jlebon** [comment]
OK, right. The generic issue of not handling `--` correctly is fixed in #1899. With that, one can do:

```
$ ostree config -- set group.my-key -1
$ cat config
[core]
repo_version=1
mode=bare

[group]
my-key=-1
```

**Comment by r4f4** [comment]
Fixed by #1899.
